test/e2e/subscription_e2e_test.go

```diff
@@ -1366,6 +1366,258 @@ var _ = Describe("Subscription", func() {
 		require.ElementsMatch(GinkgoT(), []string{"csv-a", "csv-b", "csv-e"}, got)
 	})
 
+	Context("with dependencies from different catsrc with priorities", func() {
+		var kubeClient operatorclient.ClientInterface
+		var crClient versioned.Interface
+		var crd apiextensions.CustomResourceDefinition
+		var packageMain, packageDepRight, packageDepWrong registry.PackageManifest
+		var csvsMain, csvsRight, csvsWrong []v1alpha1.ClusterServiceVersion
+		var catsrcMain, catsrcDepRight, catsrcDepWrong *v1alpha1.CatalogSource
+		var cleanup, cleanupSubscription cleanupFunc
+		const (
+			mainCSVName  = "csv-main"
+			rightCSVName = "csv-right"
+			wrongCSVName = "csv-wrong"
+		)
+
+		BeforeEach(func() {
+			kubeClient = ctx.Ctx().KubeClient()
+			crClient = ctx.Ctx().OperatorClient()
+			crd = newCRD(genName("ins"))
+		})
+
+		BeforeEach(func() {
+			packageMain = registry.PackageManifest{PackageName: genName("PkgMain-")}
+			csv := newCSV(mainCSVName, testNamespace, "", semver.MustParse("0.1.0"), nil,
+				[]apiextensions.CustomResourceDefinition{crd}, nil)
+			packageMain.DefaultChannelName = stableChannel
+			packageMain.Channels = append(packageMain.Channels, registry.PackageChannel{Name: stableChannel, CurrentCSVName: csv.GetName()})
+			csvsMain = []v1alpha1.ClusterServiceVersion{csv}
+
+			csvsRight = []v1alpha1.ClusterServiceVersion{}
+			csvsWrong = []v1alpha1.ClusterServiceVersion{}
+		})
+
+		AfterEach(func() {
+			cleanupSubscription()
```

Comment:
This could be nil, depending on the behavior of each spec, and the panic would be misleading.

Reply:
I don't think the tests hit panic or nil right? they all passed. the func value were assigned within the test.

Reply:
```suggestion
if cleanupSubscription !=nil {
     cleanupSubscription()
}
```

Reply:
> I don't think the tests hit panic or nil right? they all passed. the func value were assigned within the test.

Yup, they won't panic if they all pass. But if any of the tests end early (like if an assertion fails), the assignment to cleanupSubscription will not happen and this line will panic.

Reply:
i see